src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -156,23 +151,22 @@ class BlockingProcess(
       args
 }
 
-private object Closer {
-  def props(closeable: Closeable): Props =
-    Props(new Closer(closeable))
+private object ShutdownHook {
+  def props(op: () => Unit): Props =
+    Props(new ShutdownHook(op))
 }
 
 /*
- * Responsible for closing a closeable in order to free up any blocked
- * threads that attempt to read from the associated subtype e.g. a stream.
+ * Responsible for performing a side-effecting operation when the actor sys is required to shutdown.
  */
-private class Closer(closeable: Closeable) extends Actor {
+private class ShutdownHook(op: () => Unit) extends Actor {
 
   override def receive =
     Actor.emptyBehavior
 
   override def postStop(): Unit =
     try
-      closeable.close()
+      op()
     catch {
       case NonFatal(_) =>
```

Comment:
How will we ever find out why a shutdownHook fails? Supervision won't catch it since it fails in `postStop`.
In the case discussed previously: what if Process.destroy() is not threadsafe?